roles/matrix-server/templates/riot-web/config.json.j2

```diff
@@ -10,10 +10,24 @@
     "integrations_jitsi_widget_url": "{{ matrix_riot_web_integrations_jitsi_widget_url }}",
     "bug_report_endpoint_url": "https://riot.im/bugreports/submit",
     "enableLabs": true,
+{% if riot_roomdir_servers is defined %}
+{% for server in riot_roomdir_servers %}
+    "roomDirectory": {
+        "servers": [
+            "{{ server }}"
+        ]
+    },
+{% endfor %}
+{% else %}
     "roomDirectory": {
         "servers": [
             "matrix.org"
         ]
+{% endif %}
+{% if riot_welcome_bot|default(true)|bool %}
```

Comment:
This could also be simplified as `{% if riot_welcome_bot %}` if we have a `riot_welcome_bot` default variable specified in `roles/matrix-server/defaults/main.yml`.

By the way, naming that `matrix_riot_web_welcome_user_id` would be preferable, because:
- it groups it into the `matrix_riot_web_` "namespace", along with all other riot-web configuration
- it follows the actual riot-web configuration variable name more closely (`welcomeUserId`)